packages/debug-test/PlaygroundAutomation.ts

```diff
@@ -0,0 +1,234 @@
+				await app.waitUntil(async () => {
+					const jsEngineListItems = await $$(`${this.jsEngineComboBoxSelector}/ListItem`);
+					return jsEngineListItems.length > 1;
+				});
+
+				const jsEngineTargetListItems = await $$(`${this.jsEngineComboBoxSelector}/ListItem[@Name="${settings.jsEngine}"]`);
+				await jsEngineTargetListItems[0].click();
+
+				// wait for drop list to collapse
+				await app.waitUntil(async () => {
+					const jsEngineListItems = await $$(`${this.jsEngineComboBoxSelector}/ListItem`);
+					return jsEngineListItems.length <= 1;
+				});
+
+				testLog.message(`changed JS engine from "${oldJsEngine}" to "${await jsEngineComboBox.getText()}"`);
+			} else {
+				testLog.message(`current JS engine matches settings ("${oldJsEngine}")`);
+			}
+		} else {
+			testLog.message(`no JS engine specified, leaving at current setting ("${oldJsEngine}")`)
+		}
+
+		if (settings.hasOwnProperty('webDebugger')) {
+			await setCheckedState(webDebuggerCheckBox, settings.webDebugger);
+			restoreWebDebugger = false;
+		}
+
+		if (restoreWebDebugger) {
+			await setCheckedState(webDebuggerCheckBox, true);
+		}
+	}
+
+	private oldSettings : DebugSettings = {};
+	private newSettings : DebugSettings;
+
+	private webDebuggerCheckBoxSelector = './Window/CheckBox[@AutomationId=\"x_UseWebDebuggerCheckBox\"]';
+	private directDebuggingCheckBoxSelector = './Window/CheckBox[@AutomationId=\"x_UseDirectDebuggerCheckBox\"]';
+	private debuggerPortEditSelector = './Window/Edit[@AutomationId=\"x_DebuggerPort\"]';
+	private jsEngineComboBoxSelector = './Window/ComboBox[@AutomationId="x_JsEngine"]';
+}
+
+async function exploreUI() {
+	// const r01 = await $$('Window/Button'); // Unimplemented Command: css selector locator strategy is not supported
+	// testLog.message(`r01.length=${r01.length}`);
+
+	// const r02 = await $$('/Window/Button'); // r02.length = 0
+	// testLog.message(`r02.length = ${r02.length}`);
+
+	// const r03 = await $$('./Window/Button'); // r03.length = 5
+	// testLog.message(`r03.length = ${r03.length}`);
+	// for(const e of r03) {
+	// 	const t = await e.getText();
+	// 	testLog.message(`r03 element ${t}`);
+	// }
+	// =>
+	// r03.length = 5
+	// r03 element Minimize playground // title bar (title bar has control type Window)
+	// r03 element Maximize playground
+	// r03 element Close playground
+	// r03 element Load
+	// r03 element Unload
+
+
+	const jsEngineComboBoxes = await $$('./Window/ComboBox[@AutomationId="x_JsEngine"]');
+	testLog.message(`jsEngineComboBoxes.length ${jsEngineComboBoxes.length}`);
+
+	testLog.message("before getting text");
+	const jsEngine1 = await jsEngineComboBoxes[0].getText();
+	testLog.message("after getting text");
+	testLog.message(`jsEngineCombo before: "${jsEngine1}"`);
+
+
+	const r04 = await $$('./Window/ComboBox[@AutomationId="x_JsEngine"]/ListItem');
+	testLog.message(`r04.length = ${r04.length}`);
+	jsEngineComboBoxes[0].click(); // expand
+	await sleep(1000);
```

Comment:
Same comment as elsewhere, to avoid the hardcoded delay in the non-draft version.

Reply:
I removed the sleep call (more to the point, the helper method to explore the UI it occurred in).